### PR TITLE
Add filters and search bar to lobby menu

### LINKED ISSUE #94: Lobby List - Filter by Game Mode
From the Trello

### PR DESCRIPTION
**Pull request by T1mbits**
Added filter options and a search bar to better navigate lobbies. Also fixed some overflowing text from the meadow and story mode descriptions as well as greyed out the "about", "stats", and "unlocks" buttons.

I think this fixes issue #94 

https://github.com/user-attachments/assets/13e6a172-34a6-4c47-adb8-6d04a9d9465a

### CODE REVIEW

**Comment by T1mbits** [comment]
phanie suggested that I use checkboxes instead for the gamemode filter, but if custom game modes are going to be supported that will probably cause clutter so maybe I'll make a multi-selection dropdown menu

also I think the mode filter will need to be rewritten entirely if custom game modes are going to be supported

File: Menu/LobbySelectMenu.cs [hunk 1]
```diff
@@ -260,6 +309,70 @@ public override void Update()
...
+            public enum ModeFilter
+            {
+                All,
+                Meadow,
+                Story,
+                ArenaCompetitive
+            };
+            public enum PasswordFilter
+            {
+                [Description("Don't care")]
```
Comment (6fears7) [inline Menu/LobbySelectMenu.cs:358]: Update these please so we can have the game _attempt_ to benefit translating them like pictured below:

image

Comment (T1mbits) [reply Menu/LobbySelectMenu.cs:358]: image

like so correct?

**Comment by henpemaz** [comment]
Really good first PR. I personally don't like the placement but that can be figured out later, good to go.

**Comment by T1mbits** [comment]
do you have any suggestions  on where I can  move it to? I'm thinking about making a multi-selection dropdown for the game modes already so I could move the filter stuff  wherever you want while I'm doing that

**Comment by 6fears7** [comment]
Personally, I'd like it more centered with the  fields above. And id like to see a horizontal separator with titles so I can clearly delineate between "Create Lobby" and "Filter Lobby", since right now on that page we have both.

**Comment by 6fears7** [comment]
Id also want 

1. the larger Create button removed from the lobby selection list. 

2. Clicking a lobby joins it, instead of click to select and click separate button to join 

3. Create / join button becomes "create" and is moved to be with the creation fields 

^ these might give the space some breathing room

**Comment by T1mbits** [comment]
Do the "about", "stats", and "unlocks" buttons have any planned purpose or should I remove those too? I know in meadow mode that there are unlocks but I'm not quite sure what it would do for other game modes. Stats seems a bit weird  unless you're showing playtime/deaths/cycles in your save and about should be covered by the description underneath the mode selection

**Comment by 6fears7** [comment]
I won't speak for Henpe but I don't see the usefulness in those buttons outside of a particular game mode. I would anticipate modded modes to have descriptions inside their process menus and not lobby select

**Pull request merged**